src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -828,6 +789,64 @@ class npc_salanar_the_horseman : public CreatureScript
     }
 };
 
+class spell_deliver_stolen_horse : public SpellScriptLoader
+{
+public:
+    spell_deliver_stolen_horse() : SpellScriptLoader("spell_deliver_stolen_horse") { }
+
+    class spell_deliver_stolen_horse_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_deliver_stolen_horse_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_DELIVER_STOLEN_HORSE, SPELL_EFFECT_STOLEN_HORSE });
+        }
+
+        void HitSalanarCheck(SpellEffIndex /*effIndex*/)
+        {
+            // Can only hit Salanar due to TARGET_UNIT_NEARBY_ENTRY
+            if (Unit* salanar = GetHitUnit())
+                _salanarNear = true;
+        }
+
+        void HitTargetCheck(SpellEffIndex effIndex)
+        {
+            if (!_salanarNear)
+            {
+                PreventHitEffect(effIndex);
+                return;
+            }
+
+            if (Player* player = GetHitPlayer())
+                player->RemoveAurasDueToSpell(SPELL_EFFECT_STOLEN_HORSE);
+
+            if (Unit* caster = GetCaster())
```

Comment:
SpellScripts caster is never null

Reply:
Yes i know.. But since in every script there are a lot of checks I used them too. Same for the other. 
I can change it anyway!